modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaGetLayoutDisplayPageProviderCheck.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaGetLayoutDisplayPageProviderCheck
+	extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> importNames = javaClass.getImportNames();
+
+		if (!importNames.contains(
+				"com.liferay.layout.display.page." +
+					"LayoutDisplayPageProviderRegistry")) {
+
+			return content;
+		}
+
+		return _checkForGetLayoutDisplayPageProvider(content, javaClass);
+	}
+
+	private String _checkForGetLayoutDisplayPageProvider(
+		String content, JavaClass javaClass) {
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			Matcher matcher = _getLayoutDisplayPageProviderPattern.matcher(
+				javaMethodContent);
+
+			if (matcher.find() &&
+				hasClassOrVariableName(
+					"LayoutDisplayPageProviderRegistry", javaMethodContent,
+					content, matcher.group())) {
+
+				String line = matcher.group();
+
+				String newLine = StringUtil.replace(
+					line, "getLayoutDisplayPageProvider",
+					"getLayoutDisplayPageProviderByClassName");
+
+				javaMethodContent = StringUtil.replace(
+					javaMethodContent, line, newLine);
+			}
```

Comment:
Shouldn't this be a while loop instead, since there could be multiple instances of `getLayoutDisplayPageProvider()`? Like this:

```java
while (matcher.find()) {
	if (!hasClassOrVariableName(
		"LayoutDisplayPageProviderRegistry", javaMethodContent, content, matcher.group())) {

		continue;
	}

	String line = matcher.group();

	String newLine = StringUtil.replace(
		line, "getLayoutDisplayPageProvider",
		"getLayoutDisplayPageProviderByClassName");

	javaMethodContent = StringUtil.replace(
		javaMethodContent, line, newLine);
}
```

Reply:
Yes, I aggree.